### PR TITLE
[ACA-2214] Sharing URL being constructed from ECM Host incorrectly

### LINKED ISSUE #946: Share link not functional
**Describe the bug**
When we select a document and click on "Share" , the url provided there does not work for us.
**To Reproduce**
1. Go to  a document
2. Click on "Share" in the context menu
3. Copy the share url
4. If we browse the url, its not working. It redirects to the home page of the share. The below is the URL 
https://<<servername>>/#/preview/s/Jk6qSKsfSpiUVyOOuVNM5w

**Expected behavior**

**Screenshots**

**Desktop (please complete the following information):**

**Smartphone (please complete the following information):**

**Additional context**

### PR DESCRIPTION
**Pull request by DenysVuika**
## PR Checklist
Please check if your PR fulfills the following requirements:

```
- [x] The commit message follows our guidelines: https://github.com/Alfresco/alfresco-content-app/blob/master/CONTRIBUTING.md#commit
- [x] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)
```

## PR Type
What kind of change does this PR introduce?

```
[x] Bugfix
[ ] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Build related changes
[ ] CI related changes
[ ] Documentation content changes
[ ] Application / Infrastructure changes
[ ] Other... Please describe:
```

## What is the current behavior?

Issue Number: [ACA-2214](https://issues.alfresco.com/jira/browse/ACA-2214)

## What is the new behavior?

- fixes #946
- adds correct default for the baseShareUrl
- provides full control of the url prefix (including application route)

## Does this PR introduce a breaking change?
```
[ ] Yes
[x] No
```

### CODE REVIEW

File: src/app/app.component.spec.ts [hunk 1]
```diff
@@ -47,48 +58,59 @@ describe('AppComponent', () => {
...
-    afterEach(() => {
-      storeMock.dispatch['calls'].reset();
+  beforeEach(() => {
+    storeMock.dispatch = jasmine.createSpy('dispatch');
+  });
+
+  it('should setup baseShareUrl as per config', done => {
+    storeMock.dispatch.and.callFake((action: SetInitialStateAction) => {
+      expect(action.payload.sharedUrl).toBe(
+        'http://localhost:4200//#/preview/s/'
```
Comment (adinapitul) [inline src/app/app.component.spec.ts]: why 2 slashes ?

Comment (DenysVuika) [reply src/app/app.component.spec.ts]: Copy-pasting went wrong :)

**Comment by suzanadirla** [comment]
Is there any change in the documentation that needs to be done also?

**Comment by DenysVuika** [comment]
@suzanadirla , thanks for pointing out, I've reduced the docs as the code no longer relies on ecmHost

**Pull request merged**